### PR TITLE
magento-engcom/community-portal#1 Add TypeScript support to the project

### PR DESCRIPTION
**Pull request by XiyaMagento**
magento-engcom/community-portal#1 Add TypeScript support to the project

- Added TypeScript dependencies based on the aws-nodejs-typescript template (https://www.jamestharpe.com/serverless-typescript-getting-started/).

- Changed .js files in /functions/handlers and /functions/src to .ts files.

- serverless webpack --out dist is able to produce JS files from the TS files.

- Changed es5 to es6 in /functions/tsconfig.json

### CODE REVIEW

**Comment by XiyaMagento** [comment]
Yet to add TypeScript support for the frontend folder (having some issue transpiling jsx in TypeScript. Closing the PR for now.

File: functions/package.json [hunk 1]
```diff
@@ -5,10 +5,21 @@
   "main": "index.js",
   "dependencies": {
     "express": "^4.16.3",
-    "serverless-http": "^1.5.5"
+    "serverless": "^1.27.3",
+    "serverless-http": "^1.5.5",
+    "source-map-support": "^0.5.0"
```
Comment (vrann) [inline functions/package.json]: what is the purpose of source-map-support in our project?

File: functions/src/projects/projects.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class Project {
-    getProjects(event) {
+    getProjects(event:any) {
```
Comment (vrann) [inline functions/src/projects/projects.ts]: is there more concrete data type for lambda event? We might need to create one with the data requested from the projects. Same for the Users class

**Comment by XiyaMagento** [comment]
It's generated by webpack. I think it may not make sense to include it in the repo. Do you want me to delete it?

File: functions/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+const path = require('path');
+const slsw = require('serverless-webpack');
+
+// const entries = {};
```
Comment (vrann) [inline functions/webpack.config.js]: please avoid commented-out code

File: functions/package.json (hunk 1 above)
Comment (XiyaMagento) [reply functions/package.json]: It appears to be unnecessary. I'll proceed to remove it.

File: functions/src/projects/projects.ts (hunk 2 above)
Comment (XiyaMagento) [reply functions/src/projects/projects.ts]: It seems that events are associative arrays that can take many forms: https://docs.aws.amazon.com/lambda/latest/dg/eventsources.html

I think maybe we shouldn't specify its structure.

File: functions/webpack.config.js (hunk 3 above)
Comment (XiyaMagento) [reply functions/webpack.config.js]: Noted

File: functions/src/projects/projects.ts (hunk 2 above)
Comment (vrann) [reply functions/src/projects/projects.ts]: that's true ragrding events, but the methods of the class should not rely on the lambda eents as a paratmeters, they should rather accept an object of GetProjectRequest with the predefined fields for the project retrieving.

i.e. GitHub API: https://octokit.github.io/rest.js/#api-Issues-getComments

regarding lambda event there should be a mapper from the event  to the type (if needed)

**Comment by vrann** [comment]
regarding bundle.js my guess would be that it should not be part of the repo, but rather should be generated during deployment

**Comment by XiyaMagento** [comment]
Right. I have removed bundle.js in the latest commit.

File: functions/src/projects/projects.ts (hunk 2 above)
Comment (XiyaMagento) [reply functions/src/projects/projects.ts]: That makes sense. I'll proceed to define the structure of the input event.

**Comment by XiyaMagento** [comment]
Epic 1 completed. Closing this PR.

**Pull request closed without merging**